packages/google_maps_routes_api/lib/routes_service.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+library google_maps_routes_api;
+
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+import 'src/types/index.dart';
+
+/// A service used to calculate routes and route matrixes using
+/// Google Routes API REST endpoints.
+class RoutesService {
+  /// Creates the [RoutesService].
+  const RoutesService({required this.apiKey});
+
+  /// Google API key used for the requests.
+  final String apiKey;
+
+  static const String _routesApiUrl = 'https://routes.googleapis.com/';
+
+  /// POST https://routes.googleapis.com/directions/v2:computeRoutes
+  ///
+  /// You can provide the response field mask by using the [fields] parameter.
+  ///
+  /// Default field mask if no [fields] are given is:
+  /// 'routes.duration, routes.distanceMeters'
+  ///
+  /// You can also provide additional [headers] and [queryParams] for the
+  /// request.
+  ///
+  /// See the available URL query params and headers:
+  /// https://cloud.google.com/apis/docs/system-parameters
+  ///
+  /// Detailed documentation about how to construct the field paths:
+  /// https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/field_mask.proto
+  Future<ComputeRoutesResponse> computeRoute(
+    ComputeRoutesRequest body, {
+    String? fields,
+    Map<String, String>? headers,
+    List<String>? queryParams,
+  }) async {
+    try {
+      String url = '$_routesApiUrl/directions/v2:computeRoutes';
+      if (queryParams != null && queryParams.isNotEmpty) {
+        url += '?${queryParams.join("&")}';
+      }
+      final Map<String, String> defaultHeaders = <String, String>{
+        'X-Goog-Api-Key': apiKey,
+        'X-Goog-Fieldmask': fields ?? 'routes.duration, routes.distanceMeters',
+        'Content-Type': 'application/json',
+      };
+
+      final http.Response response = await http.post(
+        Uri.parse(url),
+        body: jsonEncode(body),
+        headers: <String, String>{...defaultHeaders, ...?headers},
+      );
+
+      if (response.statusCode != 200) {
+        throw Exception(response.body);
+      }
+
+      final ComputeRoutesResponse? result =
+          ComputeRoutesResponse.fromJson(json.decode(response.body));
+      return Future<ComputeRoutesResponse>.value(result);
+    } catch (error) {
+      rethrow;
+    }
+  }
+
+  /// POST https://routes.googleapis.com/distanceMatrix/v2:computeRouteMatrix
```

Comment:
To the first line add description of the function.
Then one empty comment line and current text.